beacon-chain/blockchain/process_attestation_helpers.go

```diff
@@ -28,6 +29,14 @@ func (s *Service) getAttPreState(ctx context.Context, c *ethpb.Checkpoint) (*sta
 		return cachedState, nil
 	}
 
+	headRoot, err := s.HeadRoot(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get head root")
+	}
+	if bytes.Equal(headRoot, c.Root) {
+		return s.HeadState(ctx)
```

Comment:
Here is a cache miss, we should update the cache

Reply:
Addressed